test/gtest/common/sort.cpp

```diff
@@ -0,0 +1,75 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "xhp-tests.hpp"
+
+using T = int;
+using DV = xhp::distributed_vector<T>;
+using LV = std::vector<T>;
+
+void test_sort(LV &v, auto func) {
+  auto size = v.size();
+  DV d_v(size);
+
+  for (std::size_t idx = 0; idx < size; idx++) {
+    d_v[idx] = v[idx];
+  }
+  barrier();
+
+  std::sort(v.begin(), v.end(), func);
+  xhp::sort(d_v, func);
+
+  barrier();
+  EXPECT_TRUE(equal(v, d_v));
+}
+
+void test_sort2s(LV &v) {
+  test_sort(v, std::less<T>());
+  test_sort(v, std::greater<T>());
+}
+
+void test_sort_randomvec(std::size_t size, std::size_t bound = 100) {
+  LV l_v = generate_random<T>(size, bound);
+  test_sort2s(l_v);
+}
+
+TEST(Sort, Random) {
+  test_sort_randomvec(1);
+  test_sort_randomvec(comm_size - 1);
+  test_sort_randomvec((comm_size - 1) * (comm_size - 1));
+  test_sort_randomvec(17);
+  test_sort_randomvec(123);
+}
+
+TEST(Sort, BigRandom) { test_sort_randomvec(1234567, 65535); }
+
+TEST(Sort, NonRandom) {
```

Comment:
putting each case into separate `TEST()` will have two benefits:
- you will see faster which test case fails
- you will execute all other tests even if one of them fails
Disadvatage is that one has to invent names for tests, but this may be also an improvement for readability,